src/ophyd_async/core/_detector.py

```diff
@@ -45,8 +45,16 @@ class DetectorTrigger(str, Enum):
 class TriggerInfo(BaseModel):
     """Minimal set of information required to setup triggering on a detector"""
 
-    #: Number of triggers that will be sent, 0 means infinite
-    number: int = Field(ge=0)
+    #: Number of triggers that will be sent, (0 means infinite) Can be:
+    #  - A single integer or
+    #  - A list of integers for multiple triggers
+    # Example for tomography: TriggerInfo(number=[2,3,100,3])
+    #:     This would trigger:
+    #:     - 2 times for dark field images
+    #:     - 3 times for initial flat field images
+    #:     - 100 times for projections
+    #:     - 3 times for final flat field images
+    number: int | list[int]
```

Comment:
I think `_total_frames` should live here. If the only reason we're putting it in the `DetectorControl` is to calculate it if `number` is a list then we can do:

``` python
    @computed_field # from pydantic
    @property
    def total_frames(self) -> int:
        return (
            sum(self.number)
            if isinstance(self.number, list) else self.number
        )

Reply:
With how many times we're doing:

``` python
       frames: int = (
            sum(trigger_info.number)
            if isinstance(trigger_info.number, list)
            else trigger_info.number
        )
```

in tests, this is seeming even more useful.